pallets/xanchor/src/lib.rs

```diff
@@ -508,24 +512,43 @@ pub mod pallet {
 			Ok(().into())
 		}
 
+		// anchorupdateProposal
 		#[pallet::weight(0)]
 		pub fn update(
 			origin: OriginFor<T>,
-			r_id: ResourceId,
-			metadata: EdgeMetadataOf<T, I>,
+			anchor_update_proposal_bytes: [u8; 82],
 		) -> DispatchResultWithPostInfo {
-			let para = ensure_sibling_para(<T as Config<I>>::Origin::from(origin))?;
-			let caller_chain_id =
-				compute_chain_id_type(u32::from(para), T::Anchor::get_chain_type());
-			let (tree_id, r_chain_id) = utils::parse_resource_id::<T::TreeId, T::ChainId>(r_id);
-			// double check that the caller is the same as the chain id of the resource
-			// also the the same from the metadata.
-			let src_chain_id: u64 = metadata.src_chain_id.try_into().unwrap_or_default();
-			let r_chain_id: u64 = r_chain_id.try_into().unwrap_or_default();
-			ensure!(
-				caller_chain_id == src_chain_id && caller_chain_id == r_chain_id,
-				Error::<T, I>::InvalidPermissions
+			ensure_sibling_para(<T as Config<I>>::Origin::from(origin))?;
+
+			// get the anchor update proposal struct from the bytes
+			let anchor_update_proposal = AnchorUpdateProposal::from(anchor_update_proposal_bytes);
+
+			let (tree_id, r_chain_id) = utils::parse_resource_id::<T::TreeId, T::ChainId>(
+				anchor_update_proposal.header().resource_id().into(),
 			);
+
+			let my_para_id = T::ParaId::get();
+			let my_chain_id =
+				utils::get_typed_chain_id_in_u64(my_para_id.try_into().unwrap_or_default());
+
+			let caller_chain_id =
+				T::ChainId::try_from(anchor_update_proposal.src_chain().chain_id())
+					.unwrap_or_default();
+
+			let typed_chain_id_of_caller: u64 =
+				utils::get_typed_chain_id_in_u64(r_chain_id.try_into().unwrap_or_default());
+
+			// verify that the chain id and the one from the parsed resource id matches
+			ensure!(my_chain_id == typed_chain_id_of_caller, Error::<T, I>::InvalidPermissions);
+
+			//construct the metadata
+			let metadata = EdgeMetadata {
+				src_chain_id: caller_chain_id,
```

Comment:
The chain id in the edge metadata should be the typed chain id (u64)

Reply:
Yes. It is the typed chain id in u64 you can take a look at line 541 to see how i derived it(`caller_chain_id`).